src/HeaderAppendPlugin.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Adds headers to the request.
+ * If the header already exists the value will be appended to the current value
```

Comment:
maybe add a line below this one saying "This only makes sense for headers that can have multiple values like 'Forwarded'". Forwarded seems the only standard request header that can have multiple values, at least as i read on https://en.wikipedia.org/wiki/List_of_HTTP_header_fields (and we could link this wikipedia article from the documentation as well)